arclightning_backend/src/router.rs

```diff
@@ -27,14 +34,41 @@ pub fn list_files(path: PathBuf) -> Result<Vec<PathBuf>, io::Error> {
 
 #[derive(Debug, Clone)]
 pub struct Router {
-    games_list: Arc<Mutex<HashMap<String, Game>>>,
+    games: Arc<Mutex<HashMap<String, Game>>>,
     static_dir: PathBuf,
+    access_key: Arc<Mutex<Option<AccessKey>>>,
+    password: String,
 }
 
 #[derive(Debug, Deserialize, Clone)]
 struct StartGameRequest {
     id: String,
 }
+#[derive(Debug, Deserialize, Clone)]
+struct PasswordRequest {
+    password: String,
+}
+
+#[derive(Debug, Clone)]
+struct AccessKey {
+    access_key: String,
+    access_time: u64,
+}
+
+impl AccessKey {
+    pub fn new(access_key: String, access_time: u64) -> Self {
+        AccessKey {
+            access_key: access_key.clone(),
+            access_time: access_time,
```

Comment:
if you're setting a field to a variable with the same name like `access_time: access_time`, you can just put `access_time`